### PR TITLE
Payment: Remove actor for collect

### LINKED ISSUE #733: PayCh.Collect: Clarify whether Collect implies the channel should be terminated
Currently, `Collect` sets `st.ToSend` to 0, implying that the PaymentChannel may continue to be used. Further submission of higher-nonce vouchers via `UpdateChannelState` will continue to increment `st.ToSend` with the same behavior as before.

However, after calling `Collect`, `Settle` no longer functions properly due to `st.SettlingAt` being nonzero. Further updates to `UpdateChannelState` will allow either party to instantly `Collect` `st.ToSend`.

#### Recommendation

Clarify semantics of `Collect`: 

* If, after `Collect`, a channel should still be usable, set `st.SettlingAt` to zero.
* If `Collect` terminates the channel, consider nullifying state in some other way - like deleting the actor.

### PR DESCRIPTION
**Pull request by aarshkshah1992**
Closes #733 .

### CODE REVIEW

**Comment by wadeAlexC** [comment]
LGTM!

File: actors/builtin/paych/paych_actor.go [hunk 1]
```diff
@@ -323,10 +323,14 @@ func (pca Actor) Collect(rt vmr.Runtime, _ *adt.EmptyValue) *adt.EmptyValue {
 	)
 	builtin.RequireSuccess(rt, codeTo, "Failed to send funds to `To`")
 
+	// even though we will delete the actor, it dosen't hurt to set this here for sanity.
 	rt.State().Transaction(&st, func() interface{} {
 		st.ToSend = big.Zero()
 		return nil
 	})
+
+	rt.DeleteActor(builtin.BurntFundsActorAddr)
```
Comment (anorth) [inline actors/builtin/paych/paych_actor.go:332]: The send to `From` above is redundant, and leaves a possibility of actually burning some funds here if we mess something up later. 

How about replacing the Send to From with setting the From as the beneficiary on deletion here? The transaction setting balance to zero is definitely redundant then.

Comment (aarshkshah1992) [reply actors/builtin/paych/paych_actor.go:332]: @anorth Why is the send to `From` redundant ?

Comment (anorth) [reply actors/builtin/paych/paych_actor.go:332]: I mean it's redundant if we set From as the beneficiary upon deletion.

**Comment by aarshkshah1992** [comment]
@anorth Please take a look.

File: actors/builtin/paych/paych_actor.go (hunk 1 above)
Comment (aarshkshah1992) [reply actors/builtin/paych/paych_actor.go]: Done.

File: actors/builtin/paych/paych_actor.go [hunk 2]
```diff
@@ -323,10 +314,8 @@ func (pca Actor) Collect(rt vmr.Runtime, _ *adt.EmptyValue) *adt.EmptyValue {
 	)
 	builtin.RequireSuccess(rt, codeTo, "Failed to send funds to `To`")
 
-	rt.State().Transaction(&st, func() interface{} {
-		st.ToSend = big.Zero()
-		return nil
-	})
+	rt.DeleteActor(st.From)
```
Comment (anorth) [inline actors/builtin/paych/paych_actor.go:317]: Great, just add a comment saying "return remaining balance to From upon deletion"

**Comment by hannahhoward** [comment]
@anorth @aarshkshah1992 clarifying something here:

the line `rt.DeleteActor(st.From)` certainly appears to delete the actor with the address `From` which is not the payment channel actor address but rather the person sending money. I'm hoping from the comment request that some how `rt.DeleteActor(st.From)` somehow means `delete this actor that's running this code, but designate the funds to the actor at address From`.

This still seems quite unintuitive if this is the case. Is there documentation somewhere of how `DeleteActor` is supposed to work?

**Comment by hannahhoward** [comment]
nm previous comment, resolved.

**Pull request merged**